src/backend/api/types.py

```diff
@@ -21,13 +23,28 @@
 
 class Bucket(BaseModel):
     Name: str
-    CreationDate: dt
+    CreationDate: Optional[dt] = None
+
+    def __eq__(self, other: Bucket) -> bool:
```

Comment:
@jecluis @irq0 
I'm not particularly happy about this `__eq__` implementation.
I've implemented a not obvious logic to cover the cases when a Bucket object is used as in parameter to a function. 
The problem is that the `CreationDate` field is always computed by the backend, so when you want to assert that the input value is equal to the return value you must discard the  `CreationDate` field in the comparison.
The logic here says that if one of the parameters in the `__eq__` function has `CreationDate` undefined, the comparison should not be performed against the `CreationDate`.
Suggestions? Is this acceptable in your opinion?

Reply:
I don't grok what you are trying to model here.

On the S3 side bucket identity is based on the name. (see https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/bucket/index.html). I guess equality based on identity makes sense, what is the creation time used for?

Is Bucket intended for S3 and the frontend? Maybe it makes sense have separate types and do more explicit conversion?

Reply:
the "problem" is that `Bucket` is an `in/out` parameter.
`CreationDate` is always set by the backend, never set by the user.
This logic attempts to compare `Bucket` depending on the context, if both A and B have `CreationDate` the comparison takes in account those field as well.